homeassistant/components/meater/sensor.py

```diff
@@ -10,10 +21,112 @@
+    if not probe.cook or probe.cook.time_remaining < 0:
+        return None
+    return dt_util.utcnow() + timedelta(probe.cook.time_remaining)
+
+
+SENSOR_TYPES = (
+    # Ambient temperature
+    MeaterSensorEntityDescription(
+        key="ambient",
+        device_class=SensorDeviceClass.TEMPERATURE,
+        name="Ambient",
+        native_unit_of_measurement=TEMP_CELSIUS,
+        state_class=SensorStateClass.MEASUREMENT,
+        available=lambda probe: probe is not None,
+        value=lambda probe: probe.ambient_temperature,
+    ),
+    # Internal temperature (probe tip)
+    MeaterSensorEntityDescription(
+        key="internal",
+        device_class=SensorDeviceClass.TEMPERATURE,
+        name="Internal",
+        native_unit_of_measurement=TEMP_CELSIUS,
+        state_class=SensorStateClass.MEASUREMENT,
+        available=lambda probe: probe is not None,
+        value=lambda probe: probe.internal_temperature,
+    ),
+    # Name of selected meat in user language or user given custom name
+    MeaterSensorEntityDescription(
+        key="cook_name",
+        name="Cooking",
+        available=lambda probe: probe is not None and probe.cook is not None,
+        value=lambda probe: probe.cook.name if probe.cook else None,
+    ),
+    # One of Not Started, Configured, Started, Ready For Resting, Resting,
+    # Slightly Underdone, Finished, Slightly Overdone, OVERCOOK!. Not translated.
+    MeaterSensorEntityDescription(
+        key="cook_state",
+        name="Cook state",
+        available=lambda probe: probe is not None and probe.cook is not None,
+        value=lambda probe: probe.cook.state if probe.cook else None,
+    ),
+    # Target temperature
+    MeaterSensorEntityDescription(
+        key="cook_target_temp",
+        device_class=SensorDeviceClass.TEMPERATURE,
+        name="Target",
+        native_unit_of_measurement=TEMP_CELSIUS,
+        state_class=SensorStateClass.MEASUREMENT,
+        available=lambda probe: probe is not None and probe.cook is not None,
+        value=lambda probe: probe.cook.target_temperature if probe.cook else None,
+    ),
+    # Peak temperature
+    MeaterSensorEntityDescription(
+        key="cook_peak_temp",
+        device_class=SensorDeviceClass.TEMPERATURE,
+        name="Peak",
+        native_unit_of_measurement=TEMP_CELSIUS,
+        state_class=SensorStateClass.MEASUREMENT,
+        available=lambda probe: probe is not None and probe.cook is not None,
+        value=lambda probe: probe.cook.peak_temperature if probe.cook else None,
+    ),
+    # Time since the start of cook in seconds. Default: 0.
+    MeaterSensorEntityDescription(
+        key="cook_time_remaining",
+        device_class=SensorDeviceClass.TIMESTAMP,
+        name="Remaining time",
+        available=lambda probe: probe is not None and probe.cook is not None,
+        value=_remaining_time_to_timestamp,
+    ),
+    # Remaining time in seconds. When unknown/calculating default is used. Default: -1
```

Comment:
This comment is confusing since a timestamp sensor should return a datetime timestamp, not an integer.

Reply:
Good point, the comment is explaining what the API returns, not what the sensor exposes, and I agree that's confusing.

Reply:
Improved in #71283